robocop/checkers/naming.py

```diff
@@ -743,3 +756,43 @@ def check_if_keyword_is_deprecated(self, keyword_name, node):
                 col=col,
                 version=f"{ROBOT_VERSION.major}.*",
             )
+
+
+class SpacesBeforeArgumentChecker(VisitorChecker):
+    reports = ("missing-4-spaces-before-argument",)
+
+    def visit_KeywordCall(self, node):
+        if node.keyword in KNOWN_KEYWORDS:
+            return
+        try:
+            known_keyword = next(keyword for keyword in KNOWN_KEYWORDS if node.keyword.startswith(f"{keyword} "))
+            self.report("missing-4-spaces-before-argument", node=node, col=len(known_keyword), keyword_name=known_keyword)
+        except StopIteration:
+            pass
+
+
+def all_builtin_keywords(suite_dir='.', src_filepath='./settings.robot'):
+    ts_builder = TestSuiteBuilder()
+    suite = ts_builder.build(suite_dir)
+    if os.path.exists(src_filepath):
+        rf_builder = ResourceFileBuilder()
+        resource = rf_builder.build(src_filepath)
+    else:
+        resource = suite.resource
+
+    settings = RobotSettings(output=None)
+    # Avoid this message: [ ERROR ] Log file is not created if output.xml is disabled.
+    OUTPUT_LOGGER.unregister_console_logger()
+    output, variables = Output(settings), VariableScopes(settings)
+    output.library_listeners.new_suite_scope()
+
+    ns = Namespace(variables, suite, resource)
+    EXECUTION_CONTEXTS.start_suite(suite, ns, output, dry_run=True)
+    ns.handle_imports()
+    for lib in ns.libraries:
+        for keyword in lib.handlers:
+            yield keyword
+    EXECUTION_CONTEXTS.end_suite()
+
+
+KNOWN_KEYWORDS = [keyword.name for keyword in all_builtin_keywords()]
```

Comment:
KNOWN_KEYWORDS should be initialized in checker init class, or even better it could be lazy loaded in ``visit_File`` (if the rule is disabled, it would never create this lookup saving as some time). By lazy loading I mean that it could create some kind of property:

```
def __init__(self):
    self._known_keywords = None

@property
def known_keywords(self):
    if self._known_keyword is None:
          self._known_keywords = get_known_keywords()
    return self._known_keywords
```

Also the scope changes for every file - one file could import for example SeleniumLibrary while the other not.. but it would be hard to track since the scope could be global or local.

Reply:
I think this is the reason why we should not include this rule into Robocop. There are many ways to dynamically import new keywords and libraries, and I am just worried that it will not be bullet-proof. Also, I think the performance of Robocop may drop, since this has quite extensive operations, in compare to the rest of Robocop rules.

What do you think @bhirsz ?

Although, this is a great candidate for a community rules, which is one of the ideas I would love to drive soon - to have a main repository of user-created rules that are helpful, but maybe not the best candidates for including them as built-in rules.